packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -344,6 +344,11 @@ class TestCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts {
       );
     }
 
+    if (!flutterProject.manifest.devDependencies.contains('flutter_test') &&
+        !flutterProject.manifest.dependencies.contains('flutter_test')) {
+      throwToolExit('flutter_test dev dependency not found!');
```

Comment:
This may want to link to a doc or be a bit more descriptive, maybe something like:

A flutter_test dev dependency is required in order to use the `flutter test` command, but no dependency was found. Please add one as follows:

```yaml
dev_dependencies:
  flutter_test:
      sdk: flutter
```

Reply:
Cool I added a link for reference to our docs and filed an issue to remove the recommendation to use either `test` or `flutter_test` in the pubspec

- https://github.com/flutter/website/issues/9931